### ISSUE TITLE
Distinguish or hide internal classes from public API documentation

### ISSUE THREAD

**Issue by 1ec5** [maplibre/maplibre-gl-js#2916]
The new documentation pages have dedicated pages for internal types such as [`Actor`](https://maplibre.org/maplibre-gl-js/docs/API/classes/maplibregl.Actor/) and [`Bucket`](https://maplibre.org/maplibre-gl-js/docs/API/interfaces/maplibregl.Bucket/) that claim to be accessible publicly via the `maplibregl` global but aren’t accessible. This documentation could be useful to contributors, but it’s misleading to downstream developers. They also clutter up the table of contents, making it more difficult for developers to browse the list to get a sense of the different customization options and integration hooks.

Similar to #2861, these pages should ideally be moved to a separate documentation set, but at least they should be labeled as private somehow.

**Comment by HarelM** [comment]
CC: @ZeLonewolf 
Let me know if you would like me to assign a bounty to this issue.
Overall adding `@hidden` to the class "main" comment should do the trick I believe.

**Comment by Wholesomebruh** [comment]
@1ec5 , so it should be permanently hidden or come with a toggle button?

**Comment by 1ec5** [comment]
Either way is fine, I guess, as long as the pages for the internal classes don’t suggest that you can access these classes publicly using `maplibregl.*`.

**Comment by drwestco** [comment]
Same for generating `maplibre-gl.d.ts`. Internal methods shouldn't be marked as public. e.g., calling these methods on `StyleLayer` without going through a `Map` instance leaves the style and the layers in an inconsistent state.
```TypeScript
	getLayoutProperty(name: string): any;
	setLayoutProperty(name: string, value: any, options?: StyleSetterOptions): void;
	getPaintProperty(name: string): unknown;
	setPaintProperty(name: string, value: unknown, options?: StyleSetterOptions): boolean;
```

**Comment by HarelM** [comment]
I don't think typescript has internal method definition so I'm not sure how to do this in the d.ts file, but adding a `@hidden` to the tsdoc comment will solve the documentation part and hopefully, people who use this method in typescript will see that it's marked as hidden...

**Comment by drwestco** [comment]
https://www.typescriptlang.org/tsconfig#stripInternal

**Comment by HarelM** [comment]
We are currently not using api extractor, it was too complicated to set up. But feel free to push this forward. We are using dts bundle generator.

**Comment by HarelM** [comment]
A list of classes that you think shouldn't be in the docs site would be a preferred solution here instead of me trying to guess what you mean :-)
I can create a PR with the addition of `@hidden` but a list of classes would really help here.
I plan to release a version shortly, which will update the docs, so the sooner the docs will be updated.

**Comment by HarelM** [comment]
Just for example - `Actor` is used by `Dispatcher` which is required in order to create a `Source`, which is also relevant to custom source, so I'm not sure if it should or shouldn't be part of the docs...

**Comment by HarelM** [comment]
Feel free to review the above linked PR.

**Comment by 1ec5** [comment]
> Just for example - Actor is used by Dispatcher which is required in order to create a Source, which is also relevant to custom source, so I'm not sure if it should or shouldn't be part of the docs...

Ah, I hadn’t thought of that. I was just comparing the current API reference to what was there before.

**Comment by HarelM** [comment]
While I don't disagree, we are generating a module, with the name maplibregl.
So all classes, even though not exported as part of the default object has this prefix.
The good part of this, is that it's automatically generated, and so if a class is missing in the API I get a warning when generating the docs, this is how I found out about the actor.
Regardless, I think it's not the same issue, or at least what the PR I opened is intended to solve...
